src/controllers/hid/hidcontroller.cpp

```diff
@@ -254,3 +295,42 @@ void HidController::sendFeatureReport(
 ControllerJSProxy* HidController::jsProxy() {
     return new HidControllerJSProxy(this);
 }
+
+QList<int> HidController::getFeatureReport(
+        unsigned int reportID) {
+    unsigned char dataRead[kReportIdSize + kBufferSize];
```

Comment:
Use a preallocated buffer as a private member variable rather than doing a heap allocation every time this is called.

Reply:
Isn't this a stack allocation? It should be faster than heap access.

Reply:
Sorry, you are correct. Pardon my C ignorance.